src/rgw/rgw_lua_background.cc

```diff
@@ -107,6 +107,31 @@ int Background::read_script() {
   return rgw::lua::read_script(&dp, lua_manager, tenant, null_yield, rgw::lua::context::background, rgw_script);
 }
 
+lua_State* Background::initialize_lguard_state(
```

Comment:
if you recreate the `lua_state_guard` maybe the function should look like that?
```suggestion
std::unique_ptr<lua_state_guard> Background::initialize_lguard_state()
```